### ISSUE TITLE
unicorn/prefer-dom-node-dataset - Undetected exception if optional chaining

### ISSUE THREAD

**Issue by gcunsolo** [sindresorhus/eslint-plugin-unicorn#2156]
Hi Unicorn, first things first: kudos for your priceless work, big fan of yours 🦄 

Got a problem tho with `unicorn/prefer-dom-node-dataset` and optional chaining

The following code should raise an exception, and yet it doesn't:
```js
document.querySelector('body')?.getAttribute('data-some-dataset');
```
Removing the optional chaining "?." raise an error as expected ;)

Thanks in advance for your time, wish you joy and rainbows

**Comment by sindresorhus** [comment]
If anyone wants to work on this, see the initial attempt in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2158.
